packages/svelte/scripts/process-messages/templates/compile-errors.js

```diff
@@ -1,54 +1,17 @@
-/** @import { Location } from 'locate-character' */
-import * as state from './state.js';
+import { CompileDiagnostic } from './utils/compile_diagnostic.js';
 
 /** @typedef {{ start?: number, end?: number }} NodeLike */
 
-export class InternalCompileError extends Error {
+export class InternalCompileError extends CompileDiagnostic {
 	name = 'CompileError';
```

Comment:
side note: I would've called this "CompilerError" (note the additional r). Is "compile error" more correct? I stumble over that mentally each time.

Reply:
The error is comming from the compiler, so I would call it a CompilerError. Not meaning there is something is wrong with the compiler, but the an error has been raised from compiler. thoughts?